src2/api/index.js

```diff
@@ -0,0 +1,78 @@
+import 'isomorphic-fetch'
+import _ from 'underscore'
+import Promise from 'bluebird'
+import querystring from 'querystring'
+import url from 'url'
+
+class Api {
+  var cache = {};
+
+  get(url) {
+    if (this.cache[url]) {
+      return Promise.resolve(this.cache[url]);
+    } else {
+      return fetch(url).then(function(response) {
+        return response.text()
+      }).then(function(text) {
+        return this.cache[url] = text
+      });
+    }
+  }
+
+  getJson(url) {
+    return this.get(url).then(JSON.parse)
```

Comment:
As above, if you use `response.json()` there you wouldn't need to parse here, or, is there a reason we need to do it like this?